app/helpers/publishers_helper.rb

```diff
@@ -33,64 +41,77 @@ def publisher_humanize_balance(publisher, currency)
     I18n.t("helpers.publisher.balance_unavailable")
   end
 
-  def next_deposit_date(today = DateTime.now)
-    if today.day > 8
-      today = today + 1.month
+  def publisher_converted_overall_balance(publisher)
+    return if publisher.default_currency == "BAT" || publisher.default_currency.blank?
+    if overall_balance = publisher.wallet &&
+       publisher.wallet.overall_balance.is_a?(Eyeshade::OverallBalance) &&
```

Comment:
Can we remove all these `is_a?` checks. Let's just go with the [nil-safe-navigation operator](https://rubyinrails.com/2017/11/17/safe-navigation-operator-ampersand-dot-in-ruby/)

```ruby
  def publisher_overall_bat_balance(publisher)
    overall_ballance = publisher.wallet&.overall_balance
    if overall_balance&.amount_default_currency.present?
      I18n.t(
        "helpers.publisher.balance_pending_approximate",
        amount: '%.2f' % overall_balance.amount_default_currency,
        code: overall_balance.default_currency
      )
    else 
      I18n.t("helpers.publisher.conversion_unavailable", code: publisher.default_currency)
    end
  end
```